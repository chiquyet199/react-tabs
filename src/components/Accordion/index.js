import Accordion from "./Accordion";
import AccordionWithCount from "./AccordionWithCount";
import AccordionWithIconTitle from "./AccordionWithIconTitle";

export {Accordion, AccordionWithCount, AccordionWithIconTitle};
