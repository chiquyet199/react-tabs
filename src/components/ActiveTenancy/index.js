import DepositFree from "./DepositFree";
import Housemates from "./Housemates";
import TenancyTerms from "./TenancyTerms";

export {DepositFree, Housemates, TenancyTerms};
