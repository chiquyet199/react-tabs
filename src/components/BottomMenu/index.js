import {connect} from "react-redux";
import {go} from "../../actions/navigation";
import BottomMenu from "./BottomMenu";
import MenuItems from "../../schemas/Menu";

const mapStateToProps = state => ({
  items: MenuItems,
  location: state.router.location.pathname,
});

const mapDispatchToProps = {
  go,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(BottomMenu);
