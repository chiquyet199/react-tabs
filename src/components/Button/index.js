import React from "react";
import {View} from "react-native";
import {
  array,
  bool,
  func,
  object,
  oneOf,
  oneOfType,
  number,
  string,
} from "prop-types";
import {
  ButtonContainer,
  Left,
  Middle,
  Right,
  BtnText,
  StyledIcon,
  IconContainer,
  MenuButtonIconContainer,
  BounceContainer,
  MenuButtonText,
  ButtonIcon,
  TabContainer,
  TabText,
} from "./Button";
import Icon from "../Icon";
import {applyStyleObjectToStyledComponent, themed} from "../../utils/common";

export const Button = ({
  left,
  right,
  middle,
  onClick,
  type,
  noborder,
  style,
  iconsize,
  theme,
  disabled,
}) => {
  const color = theme && theme.buttons[type].text;

  return (
    <ButtonContainer
      noborder={noborder}
      type={type}
      onPress={onClick}
      activeOpacity={0.6}
      style={style}
      theme={theme}
      disabled={disabled}
    >
      {left && (
        <Left size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={left}
            size={iconsize || 32}
            color={color}
          />
        </Left>
      )}
      {middle && (
        <Middle lefticon={left} righticon={right} size={iconsize || 32}>
          <BtnText ellipsizeMode="tail" type={type} theme={theme}>
            {middle}
          </BtnText>
        </Middle>
      )}
      {right && (
        <Right size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={right}
            size={iconsize || 32}
            color={color}
          />
        </Right>
      )}
    </ButtonContainer>
  );
};

Button.propTypes = {
  left: oneOfType([string, bool]),
  middle: string,
  right: oneOfType([string, bool]),
  onClick: func,
  type: string,
};

Button.defaultProps = {
  left: null,
  right: null,
  type: "primary",
};

export const IconButton = ({
  onClick,
  name,
  type,
  style,
  iconsize,
  bias,
  theme,
}) => (
  <IconContainer style={style} type={type} onPress={onClick}>
    <Icon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={theme && theme.buttons[type].text}
    />
  </IconContainer>
);

IconButton.propTypes = {
  name: string.isRequired,
  onClick: func,
  type: string.isRequired,
};

const ThemedIconContainer = ({
  name,
  type,
  style,
  iconsize,
  bias,
  theme: {buttons},
}) => (
  <IconContainer size={iconsize} style={style} type={type}>
    <ButtonIcon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={buttons[type].text}
    />
  </IconContainer>
);

ThemedIconContainer.propTypes = {
  name: string.isRequired,
  type: string.isRequired,
  style: oneOf([array, object]),
  iconsize: number,
  bias: number,
  theme: object.isRequired,
};

export const StyledIconContainer = themed(ThemedIconContainer);

export const MenuButton = ({
  onClick,
  name,
  type,
  iconContainerStyle,
  children,
  iconsize,
  theme,
}) => {
  const StyledInnerContainer = applyStyleObjectToStyledComponent(
    iconContainerStyle,
    MenuButtonIconContainer,
  );

  return (
    <View style={{flex: 1, alignItems: "center"}}>
      <StyledInnerContainer type={type} noBorder onPress={onClick}>
        <Icon
          type={type}
          name={name}
          size={iconsize || 32}
          color={theme.buttons[type].text}
        />
      </StyledInnerContainer>
      <MenuButtonText type={type}>{children}</MenuButtonText>
    </View>
  );
};

MenuButton.propTypes = {
  onClick: func,
  name: string,
  type: string,
  iconContainerStyle: object,
  iconsize: number,
  theme: object,
};

export const FormBounceButton = ({children, onClick, globaltheme, theme}) => (
  <BounceContainer>
    <Button
      onClick={onClick}
      middle={children}
      type={theme[globaltheme].form.bounceButton}
      noBorder
    />
  </BounceContainer>
);

FormBounceButton.propTypes = {
  children: string.isRequired,
  globaltheme: oneOf(["dark", "light"]),
  onClick: func.isRequired,
};

export const TabButton = ({onClick, left, iconsize, middle, type}) => (
  <TabContainer type={type} onPress={onClick}>
    <StyledIcon type={type} name={left} size={iconsize || 20} color="inherit" />
    <TabText ellipsizeMode="tail" type={type}>
      {middle}
    </TabText>
  </TabContainer>
);

TabButton.propTypes = {
  left: string,
  type: string.isRequired,
  middle: string.isRequired,
  iconsize: number,
  onClick: func,
};

export default themed(Button);
