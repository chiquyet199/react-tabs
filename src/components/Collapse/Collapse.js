import React, {Component} from "react";
import {View, TouchableOpacity, Animated, StyleSheet} from "react-native";
import {colors} from "../../constants/theme";
import {func, bool} from "prop-types";
import {isWeb} from "../../utils/common";

export class Collapse extends Component {
  state = {
    animation: new Animated.Value(),
    minHeight: 0,
    maxHeight: 0,
  };

  shouldComponentUpdate(nextProps) {
    return nextProps.collapsed !== this.props.collapsed;
  }

  _onToggle = () => {
    const {animation} = this.state;
    const {collapsed, onToggle} = this.props;
    const initialValue = collapsed
      ? this.state.minHeight
      : this.state.maxHeight + this.state.minHeight;

    const finalValue = collapsed
      ? this.state.maxHeight + this.state.minHeight
      : this.state.minHeight;

    animation.setValue(initialValue);
    Animated.spring(animation, {
      toValue: finalValue,
    }).start();
    onToggle();
  };

  _setMinHeight = event => {
    this.setState({
      minHeight: event.nativeEvent.layout.height,
    });
  };

  _setMaxHeight = event => {
    this.setState({
      maxHeight: event.nativeEvent.layout.height,
    });
    const {animation, minHeight} = this.state;
    const {collapsed} = this.props;
    if (collapsed && isWeb) {
      animation.setValue(minHeight);
      Animated.spring(animation, {
        toValue: minHeight,
      }).start();
    }
  };

  render() {
    let header = null;
    let body = null;
    React.Children.forEach(this.props.children, child => {
      if (child.props.type === "header") {
        header = (
          <TouchableOpacity onPress={this._onToggle}>{child}</TouchableOpacity>
        );
      } else if (child.props.type === "body") {
        body = child;
      }
    });

    return (
      <Animated.View
        {...this.props}
        style={[styles.container, {height: this.state.animation}]}
      >
        <View onLayout={this._setMinHeight}>{header}</View>
        <View onLayout={this._setMaxHeight}>{body}</View>
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.white,
    marginTop: 10,
    overflow: "hidden",
  },
});

Collapse.propTypes = {
  collapsed: bool.isRequired,
  onToggle: func.isRequired,
};

export default Collapse;
