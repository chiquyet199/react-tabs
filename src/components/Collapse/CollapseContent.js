import React from "react";
import styled from "styled-components/native";

export const CollapseContentContainer = styled.View`
  ${({vpadding}) =>
    vpadding && `padding-top: ${vpadding}px; padding-bottom: ${vpadding}px`};
  ${({hpadding}) =>
    hpadding && `padding-left: ${hpadding}px; padding-right: ${hpadding}px;`};
  ${({gpadding}) => gpadding && `padding: ${gpadding}px`};
`;

export const CollapseContent = ({children, ...otherProps}) => (
  <CollapseContentContainer {...otherProps}>
    {children}
  </CollapseContentContainer>
);
