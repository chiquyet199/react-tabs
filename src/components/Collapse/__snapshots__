/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Collapse/__native_snapshots__