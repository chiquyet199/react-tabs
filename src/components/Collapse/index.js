export {Collapse} from "./Collapse";
export {CollapseContent} from "./CollapseContent";
