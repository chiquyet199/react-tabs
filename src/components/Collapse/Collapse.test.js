import React from "react";
import {shallow} from "enzyme";
import {Collapse, CollapseContent} from "../../components/Collapse";
import {Platform, Text, View} from "react-native";
import configureStore from "redux-mock-store";
import {initialState} from "../../__mocks__/initialState";

const mockProps = {
  onToggle: jest.fn(),
  collapsed: true,
};

const mockStore = configureStore(mockProps);

describe("Collapse", () => {
  let wrapper;
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(
      <Collapse {...mockProps}>
        <CollapseContent padding="10px 0" type="header">
          <Text>Test text header</Text>
        </CollapseContent>
        <CollapseContent padding="10px" type="body">
          <Text>Test text body</Text>
        </CollapseContent>
      </Collapse>,
      {
        context: store,
      },
    );
  });

  it("renders proper snapshot - " + Platform.OS, () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has passed props to Collapse component", () => {
    const collapsedProp = wrapper.props().collapsed;
    expect(collapsedProp).toEqual(mockProps.collapsed);
    const onToggleProp = wrapper.props().onToggle;
    expect(onToggleProp).toBe(mockProps.onToggle);
  });

  it("calls onToggle function when clicked", () => {
    wrapper.props().onToggle(); // trigger
    expect(mockProps.onToggle).toHaveBeenCalled();
  });

  it("calls onToggle function when component mounts", () => {
    expect(mockProps.onToggle).toHaveBeenCalled();
  });

  describe("Collapse Content", () => {
    it("has onLayout function", () => {
      const views = wrapper.find(View); // trigger
      views.forEach(view => {
        expect(view.props().onLayout).toHaveLength(1);
        expect(view.prop("onLayout")).toBeInstanceOf(Function);
      });
    });
    it("renders CollapseContent", () => {
      const contents = wrapper.find(CollapseContent);
      expect(contents).toHaveLength(2);
      contents.forEach(content => {
        expect(content.find(Text)).toHaveLength(1);
        if (content.prop("type") === "header") {
          expect(content.prop("padding")).toEqual("10px 0");
          expect(content.find(Text).props().children).toEqual(
            "Test text header",
          );
        }
        if (content.prop("type") === "body") {
          expect(content.prop("padding")).toEqual("10px");
          expect(content.find(Text).props().children).toEqual("Test text body");
        }
      });
    });
  });
});
