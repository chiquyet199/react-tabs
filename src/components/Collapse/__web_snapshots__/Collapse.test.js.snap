// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders proper snapshot - web 1`] = `
<AnimatedComponent
  collapsed={true}
  onToggle={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": AnimatedValue {
        "_animation": null,
        "_children": Array [],
        "_listeners": Object {},
        "_offset": 0,
        "_value": undefined,
      },
      "marginTop": 10,
      "overflow": "hidden",
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      onPress={[Function]}
    >
      <CollapseContent
        padding="10px 0"
        type="header"
      >
        <Text>
          Test text header
        </Text>
      </CollapseContent>
    </TouchableOpacity>
  </View>
  <View
    onLayout={[Function]}
  >
    <CollapseContent
      padding="10px"
      type="body"
    >
      <Text>
        Test text body
      </Text>
    </CollapseContent>
  </View>
</AnimatedComponent>
`;
